ts/src/gate.ts

```diff
@@ -5028,7 +5112,14 @@ export default class gate extends Exchange {
         } else {
             request['leverage'] = leverage.toString ();
         }
-        const response = await this[method] (this.extend (request, query));
+        let response = undefined;
+        if (market['type'] === 'swap') {
```

Comment:
we can use `market['swap']` here directly

Reply:
@carlosmiei how about update all type checking for swap, future, option to `market[key]`?